File: node/wags-media-repository-api/src/models/book.ts
export type BookSeries = {
    bookSeriesId: number;
    name: string;
    colorCode: string;
    bookCount?: number;
}

export type BookGenre = {
    bookGenreId: number;
    name: string;
    colorCode: string;
    bookCount?: number;
}

export type BookFormat = {
    bookFormatId: number;
    name: string;
    colorCode: string;
    bookCount?: number;
}

export type BookStatus = {
    bookStatusId: number;
    name: string;
    colorCode: string;
}

export type BookType = {
    bookTypeId: number;
    name: string;
    colorCode: string;
}

export type Book = {
    bookId: number;
    bookStatusId: number;
    bookTypeId: number;
    bookSeriesId?: number;
    title: string;
    subTitle: string;
    fullTitle?: string;
    author: string;
    link: string;
    dateStarted?: Date;
    dateCompleted?: Date;
    rating: number;
    thoughts: string;
    bookNotesUrl?: string;
    coverImageUrl: string;
    currentPage: number;
    pageCount: number;
    progress: number;
    sortOrder?: number;
    isAtLibrary: boolean;
    isPurchased: boolean;
    genres: BookGenre[];
    formats: BookFormat[];
    status?: BookStatus;
    type?: BookType;
    series?: BookSeries;
}

export type BookGenreLink = {
    bookId: number;
    genreId: number;
    genreName: string;
    genreColorCode: string;
}

export type BookFormatLink = {
    bookId: number;
    formatId: number;
    fomatName: string;
    formatColorCode: string;
}

File: tools/WagsMediaGenerator/Models/Book.cs
namespace WagsMediaGenerator.Models;

public class Book
{
    public int BookId { get; set; }

    public string Title { get; set; } = string.Empty;
    
    public string SubTitle { get; set; } = string.Empty;
    
    public string FullTitle { get; set; } = string.Empty;
    
    public string Author { get; set; } = string.Empty;
    
    public string Link { get; set; } = string.Empty;
    
    public DateTime? DateStarted { get; set; }
    
    public DateTime? DateCompleted { get; set; }
    
    public int Rating { get; set; }

    public string BookNotesUrl { get; set; } = string.Empty;
    
    public string Thoughts { get; set; } = string.Empty;
    
    public string CoverImageUrl { get; set; } = string.Empty;
    
    public bool IsAtLibrary { get; set; }
    
    public bool IsPurchased { get; set; }

    public int CurrentPage { get; set; } = 1;

    public int PageCount { get; set; }

    public int Progress { get; set; }
    
    public int? SortOrder { get; set; }

    public Tag Status { get; set; } = new();

    public Tag Type { get; set; } = new();
    
    public Tag? Series { get; set; }

    public IReadOnlyCollection<Tag> Genres { get; set; } = [];
    
    public IReadOnlyCollection<Tag> Formats { get; set; } = [];
}